pkg/kopia/stream.go

```diff
@@ -36,18 +36,28 @@ const (
 	pullRepoPurpose = "kando location pull"
 )
 
+// SnapshotInfo tracks kopia snapshot information produced by a kando command in a phase
+type SnapshotInfo struct {
+	// BackupID is the snapshot ID produced by kopia snapshot operation
+	BackupID string `json:"backupID"`
+	// LogicalSize is the sum of cached and hashed file size in bytes
+	LogicalSize string `json:"logicalSize"`
+	// LogicalSize is the uploaded size in bytes
+	PhysicalSize string `json:"physicalSize"`
+}
+
 // Write creates a kopia snapshot from the given reader
 // A virtual directory tree rooted at filepath.Dir(path) is created with
 // a kopia streaming file with filepath.Base(path) as name
-func Write(ctx context.Context, path string, source io.Reader) (string, string, error) {
+func Write(ctx context.Context, path string, source io.Reader) (string, error) {
```

Comment:
I'd suggest keeping the amount of time that this struct is serialized as short as possible - can the caller perform the serialization when needed as output?